### PR TITLE
docs: Reorg compiler docs in book

### PR DESCRIPTION
**Pull request by max-sixty**

### CODE REVIEW

File: web/book/src/integrations/cli.md
Comment (eitsupi) [inline web/book/src/integrations/cli.md:1]: Could you include prqlc README?

Comment (max-sixty) [reply web/book/src/integrations/cli.md:1]: Sorry, this was an automatically added file by mdbook, I'll fix

File: web/book/src/SUMMARY.md [hunk 1]
```diff
@@ -74,7 +67,7 @@
 
 - [Integrations](./integrations/README.md)
 
-  - [CLI](./integrations/cli.md)
+  - [`prqlc`](./integrations/prqlc.md)
```
Comment (max-sixty) [inline web/book/src/SUMMARY.md]: What should this be? I think it is / will be referred to as `prqlc`, but maybe it should be more obvious for folks who are less used to compilers?

Comment (aljazerzen) [reply web/book/src/SUMMARY.md]: "prqlc CLI"?

File: web/book/src/SUMMARY.md [hunk 2]
```diff
@@ -88,16 +81,18 @@
...
+    - [Helpers](./compiler/internals/helpers.md)
+    - [Name resolving](./compiler/internals/name-resolving.md)
+    - [Functions](./compiler/internals/functional-lang.md)
+    - [Syntax highlighting](./compiler/internals/syntax-highlighting.md)
 
-- [Internals](./internals/README.md)
+- [Contributing to PRQL](./contributing/README.md)
 
-  - [Compiler architecture](./internals/compiler-architecture.md)
-  - [Helpers](./internals/helpers.md)
```
Comment (aljazerzen) [inline web/book/src/SUMMARY.md:98]: Helpers could be lifted out of internals?

**Review by aljazerzen — COMMENTED** [review]
Much better

File: web/book/src/SUMMARY.md (hunk 1 above)
Comment (max-sixty) [reply web/book/src/SUMMARY.md]: Funny you should suggest, check the latest :)

File: web/book/src/SUMMARY.md (hunk 2 above)
Comment (max-sixty) [reply web/book/src/SUMMARY.md:98]: Good idea, done

File: prql-compiler/prqlc/README.md [hunk 3]
```diff
@@ -1,7 +1,68 @@
 # PRQL compiler CLI — `prqlc`
 
-`prqlc` is a CLI for PRQL compiler; A single, dependency-free binary that
-compiles PRQL into SQL.
+`prqlc` serves as a CLI for the PRQL compiler. It is a single, dependency-free
```
Comment (max-sixty) [inline prql-compiler/prqlc/README.md:3]: I moved this section up, and put some of it through GPT4 to reword it — recommended!

**Pull request merged**